lib/t/feeds.ex

```diff
@@ -85,23 +85,46 @@ defmodule T.Feeds do
   ### Feed
 
   @type feed_cursor :: DateTime.t()
-  @type feed_profile :: {%FeedProfile{}, distance_km :: non_neg_integer}
+  @type feed_profile :: %FeedProfile{}
 
   @spec fetch_feed(
           Ecto.UUID.t(),
           Geo.Point.t(),
           String.t(),
           [String.t()],
           pos_integer,
+          pos_integer,
+          pos_integer,
+          pos_integer,
           feed_cursor | nil
         ) ::
           {[feed_profile], feed_cursor}
-  def fetch_feed(user_id, location, gender, gender_preferences, count, feed_cursor) do
+  def fetch_feed(
+        user_id,
+        location,
+        gender,
+        gender_preferences,
+        min_age,
+        max_age,
+        distance,
```

Comment:
too many arguments, easy to make a typo, try using some collection like a list or a map or maybe even a struct to put them in, `def fetch_feed(user_id, location, gender, gender_preference: gender_preference, min_age: min_age, max_age: max_age, distance: distance)`

Reply:
```elixir
@spec fetch_feed(Ecto.UUID.t(), Geo.Point.t(), String.t, Keyword.t, ...etc) :: {[feed_profile], feed_cursor}
def fetch_feed(user_id, location, gender, filters, count, feed_cursor) do
  gender_preference = filters[:gender_preference] || raise "missing gender preference filter"
  min_age = filters[:min_age] || raise "missing min age filter"
  # etc.
  continue_feed(..., filters, ...)
end 
```

Reply:
Since I later saw that filters are optional, there is no need to `raise` in case they are missing and `filters` can be passed unchecked to `continue_feed` where filters can be passed to each `maybe_filter` function and apply where clauses only if `if max_age = filters[:max_age] do`.